autoptsserver.py

```diff
@@ -623,6 +632,10 @@ def shutdown_pts_bpv(self):
     autoptsservers = []
     server_count = len(_args.srv_port)
     finish_count = CounterWithFlag(init_count=0)
+    for i in range(server_count):
+        if not _com_port_exists(_args.dongle[i]):
+            sys.exit(f"Port {_args.dongle[i]} not found")
```

Comment:
I'm unsure whether this is the best place/way to do this. 

I'm not a Python expert, but the `_args.dongle` is effectively used to instantiate a PyPTS object with `device = args.dongle`. 

Given that the class allows `None` as the type, it seems that it's possible to actually instantiate a PyPTS without a dongle. Perhaps if the dongle is inserted later? 

Perhaps this check should be moved to `PyPTS` and add a check when instantiate a `PyPTS` with `device != None` where `_com_port_exists` can be called. This could then be an internal method for the class, e.g.

```python
if device is not None and not self._com_port_exists(device):
    raise ValueError(...)
```

or something similar to that

Reply:
Made some changes. It should only check the dongles arguments and not server.

Reply:
> However, I'd want the check here to ensure it is caught before the servers are started.

One argument for moving it to the place that actually uses them, is that this check will only be applied if `autoptsserver.py` is run as the main script, whereas moving the check would apply the check to any uses of `PyPTS`

Reply:
That's also a good argument. Maybe we can add it to both places then?

Reply:
@sjanc what do you think?